### ISSUE TITLE
StaticFiles with .md extensions cause NoMethod error

### ISSUE THREAD

**Issue by janpio** [benbalter/jekyll-relative-links#33]
I have a project where I wanted to try the collections feature of this plugin. Unfortunately switching `collections` to `true` breaks `jekyll s`:

First run was this configuration:
```
relative_links:
  enabled: true
  collections: false
```
Result:
```
sujan@LenovoX1:/mnt/c/Users/Jan/Documents/ionic.zone$ bundle exec jekyll s
Configuration file: /mnt/c/Users/Jan/Documents/ionic.zone/_config.yml
       Deprecation: You appear to have pagination turned on, but you haven't included the `jekyll-paginate` gem. Ensure
you have `plugins: [jekyll-paginate]` in your configuration file.
            Source: /mnt/c/Users/Jan/Documents/ionic.zone
       Destination: /mnt/c/Users/Jan/Documents/ionic.zone/_site
 Incremental build: disabled. Enable with --incremental
      Generating...
                    done in 4.694 seconds.
                    Auto-regeneration may not work on some Windows versions.
                    Please see: https://github.com/Microsoft/BashOnWindows/issues/216
                    If it does not work, please upgrade Bash on Windows or run Jekyll with --no-watch.
 Auto-regeneration: enabled for '/mnt/c/Users/Jan/Documents/ionic.zone'
    Server address: http://127.0.0.1:4000/
  Server running... press ctrl-c to stop.
^C
```

Second, failing, run was done with this config:
```
relative_links:
  enabled: true
  collections: false
```
Result:
```
sujan@LenovoX1:/mnt/c/Users/Jan/Documents/ionic.zone$ bundle exec jekyll s
Configuration file: /mnt/c/Users/Jan/Documents/ionic.zone/_config.yml
       Deprecation: You appear to have pagination turned on, but you haven't included the `jekyll-paginate` gem. Ensure
you have `plugins: [jekyll-paginate]` in your configuration file.
            Source: /mnt/c/Users/Jan/Documents/ionic.zone
       Destination: /mnt/c/Users/Jan/Documents/ionic.zone/_site
 Incremental build: disabled. Enable with --incremental
      Generating...
jekyll 3.5.2 | Error:  undefined method `content' for #<Jekyll::StaticFile:0x0000000001ea5320>
```
From the error message I assume that somehow this line is failing:
https://github.com/benbalter/jekyll-relative-links/blob/b7d89c7cf7de7ccf15c25956a49639d8dcc8debf/lib/jekyll-relative-links/generator.rb#L43 Probably because of some malformed .md file or link I have in this project.

How can I debug this further?  
What should I do to find out which file is causing this?

**Comment by benbalter** [comment]
This should be fixed via 137f4b8. 

The issue is that you likely have a `.md` file without YAML front matter someplace in your site. The Jekyll::StaticFile model doesn't have a `content` method, which is causing the error. In https://github.com/benbalter/jekyll-relative-links/blob/b7d89c7cf7de7ccf15c25956a49639d8dcc8debf/lib/jekyll-relative-links/generator.rb#L35 we check to ensure we're only looking at Markdown file, but didn't account for Markdown files that were static files, which we should. 

Thanks for the detailed bug report!

**Comment by janpio** [comment]
> The issue is that you likely have a .md file without YAML front matter someplace in your site. 

Yep, when I start a file I just collect info and text bits in it and only later add the front matter.

Thanks for fixing this so quickly, will look at it again as soon as it is released.
